worlds/psychonauts/Options.py

```diff
@@ -0,0 +1,156 @@
+    default = False
+
+
+class StartingMentalMagnet(Toggle):
+    """Start with Mental Magnet, pulling health, ammo, and arrowhead drops towards you."""
+    display_name = "Start with Mental Magnet"
+    default = True
+
+
+class StartingCobwebDuster(Toggle):
+    """Start with the Cobweb Duster."""
+    display_name = "Start with Cobweb Duster"
+    default = False
+
+
+class RandomStartingMinds(Range):
+    """Start with a random number of mind/area unlocking items."""
+    display_name = "Random Starting Minds"
+    range_start = 0
+    range_end = 10
+    default = 3
+
+
+class LootboxVaults(Toggle):
+    """Turns all Memory Vaults into Lootboxes! Vaults will contain 10-50 Arrowheads, with a 50% chance to recieve a rank up.
+    If you get really lucky, you can even win jackpots of up to 250 Arrowheads or Five Ranks Up!
+    If False, Vaults will always give One Rank Up and 15 Arrowheads."""
+    display_name = "Lootbox Vaults"
+    default = True
+
+class FasterLindaBoss(Toggle):
+    """Make the Linda Boss chase sequences faster by increasing the speed of the air bubble."""
+    display_name = "Faster Linda Boss"
+    default = True
+
+class EasyMillaRace(Toggle):
+    """Make the race in Milla's Mind easier by removing Bobby from the race, and increasing player speed to 1.5x"""
+    display_name = "Easy Milla Race"
+    default = False
+
+
+class EasyFlightMode(Toggle):
+    """Replicate the Levitation Flight Glitch by simply holding down float in the air. No more button mashing.
+    WARNING: Makes getting locations out of logic VERY easy."""
+    display_name = "Easy Flight Mode"
+    default = False
+
+
+class EnemyDamageMultiplier(Range):
+    """Adjust how much damage Raz takes from enemies for an additional challenge.
+    If set to 0, Raz will take no damage from enemies, but will still flinch."""
+    display_name = "Enemy Damage Multiplier"
+    range_start = 0
+    range_end = 5
+    default = 1
+
+
+class InstantDeathMode(Toggle):
+    """Take any amount of damage, die instantly. Has priority over Enemy Damage Multiplier."""
+    display_name = "Instant Death Mode"
+    default = False
+
+
+class Goal(Choice):
+    """Win Condition
+    Asylum Brain Tank: Climb the Asylum and defeat the Coach Oleander Brain Tank.
+
+    Brain Hunt: Find the required amount of Camper Brains and re-brain them in Ford's Sanctuary.
+
+    Asylum Brain Tank and Brain Hunt: Find the required amount of Camper Brains, AND defeat the Coach Oleander Brain Tank."""
+    display_name = "Goal"
+    option_braintank = 0
+    option_brainhunt = 1
+    option_braintank_and_brainhunt = 2
+    default = 0
+
+
+class BrainsRequired(Range):
+    """Number of Brains required to collect to Win, or open final door to Meat Circus Final Bosses.
+
+    If Goal is not Brain Hunt, or Asylum Brain Tank and Brain Hunt, this does nothing."""
+    display_name = "Brains Required"
+    range_start = 1
+    range_end = 19
+    default = 10
+
+
+class RequireMeatCircus(Toggle):
```

Comment:
I might be missing something, but I think this option could be incorporated directly in the Goal option if it was an `OptionSet` instead of a `Choice`.

Reply:
The KH2 AP implementation uses the same method to control the Final Ansem requirement for goal. If there were more goal options than just Brain Tank, Brain Hunt, and Meat Circus, I would consider it. This would also take sizable rewrites to goal logic, and making sure that ONLY Meat Circus still works as a standalone goal, as well as deciding what to do if none are selected. The base randomizer mod is also structured around allowing access to the bosses only if all other goals are completed.

Reply:
So I was missing something. This make complete sense considering your explanation 👍